locked-asset/energy-factory/src/unlock_with_penalty.rs

```diff
@@ -91,8 +91,23 @@ pub trait UnlockWithPenaltyModule:
         let new_unlock_epoch = current_epoch + reduce_result.new_lock_epochs;
 
         let unlocked_tokens = reduce_result.unlocked_tokens;
+        let penalty_amount = &payment.amount - &unlocked_tokens.amount;
         let new_locked_tokens = self.lock_tokens(unlocked_tokens, new_unlock_epoch);
 
+        let amount_to_burn = &payment.amount - &penalty_amount;
```

Comment:
so all of this was missing and was an error here ? does burn_penalty resolve the sending of 50% out of that to the fees_collector ?

Reply:
No, previously, we were doing this in the common function, but we can't do that anymore. For `unlockEarly`, we don't have to burn the locked token, in case user wants to cancel unbond. For `reduceLockPeriod`, we do have to burn the old token, as we mint a new one with the new lock period.